### PR TITLE
Allow a function to be passed for allowSafe

### PR DESCRIPTION
**Pull request by ryanbrunner**
Allows the passing of a function for allowSafe, for when whether requests
should be considered safe is conditional based on the request or the response.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -37,8 +37,9 @@ function corsGate(options) {
     const origin = (req.headers.origin || '').toLowerCase().trim();
 
     if (!origin) {
+      const allowSafe = typeof(options.allowSafe) === 'boolean' ? options.allowSafe : options.allowSafe(req, res);
```
Comment (skeggse) [inline index.js]:
```suggestion
      const allowSafe = !!(typeof options.allowSafe === 'function' ? options.allowSafe(req) : options.allowSafe);
```

File: test/cors-gate.spec.js [hunk 2]
```diff
@@ -126,6 +126,20 @@ describe('cors-gate', function() {
         .get('/get')
         .expect(403, done);
     });
+
+    it('can be given a function', function(done) {
+      this.app.use(corsGate({
+        strict: true,
+        allowSafe: () => false,
```
Comment (skeggse) [inline test/cors-gate.spec.js]: probably test `() => true` too, right?

File: index.js (hunk 1 above)
Comment (ryanbrunner) [reply index.js]: Was the removal of `res` intentional in your suggestion? We actually use `res` in the one case where we use this function.

Comment (skeggse) [reply index.js]: rats; yeah, it was

File: test/cors-gate.spec.js (hunk 2 above)
Comment (ryanbrunner) [reply test/cors-gate.spec.js]: We could. I figured the logic of `allowSafe` itself is well tested - we just want to ensure that it supports calling a function in the first place.

**Pull request merged**